### DISCUSSION TITLE
Dataviewjs Incorporating Templater - Replace Prompt w/ Suggester

### DISCUSSION THREAD

**Discussion by julroger2013** [SilentVoid13/Templater#1234] [Help]
Hi There - 

I found the following dataviewjs script online. It draws on the following plugins: Buttons, MetaEdit, Templater

```dataviewjs
const {createButton} = app.plugins.plugins["buttons"];
const {update} = app.plugins.plugins['metaedit'].api;

const defer = async (file, key) => {
const status = await app.plugins.plugins['templater-obsidian'].templater.functions_generator.internal_functions.modules_array[4].static_functions.get('prompt')("Status?");
await update(key, status, file);
};

dv.table(
["Name", "Status", "Last Modified", "Workspace", "Button"],
dv.pages('"<file path>"').filter(t => t['status'] == "<designated status>").map(t => [
t.file.link,
t['status'],
t['workspace-link'],
t.file.mtime.toFormat("yyyy.MM.dd - hhmm"),
createButton({
app,
el: this.container,
args: { name: "Change Status" },
clickOverride: { click: defer, params: [t.file.path, 'status'] }
})
])
)
```

Its function is to create a table of files fitting two parameters: 1) Located in a designated directory; 2) Having a specified value for the field "status"

Each row of the table includes a "Change Status" button. The current version triggers a text input box via Templater, and the input into that box is then used to update the "status" field in the file associated with that row of the table. 

I'm wanting to modify this portion of the code, which I think draws exclusively on Templater:

```
const defer = async (file, key) => {
const status = await app.plugins.plugins['templater-obsidian'].templater.functions_generator.internal_functions.modules_array[4].static_functions.get('prompt')("Status?");
await update(key, status, file);
};
```
Instead of getting an input box, where I type in the desired status, I'd like to pull up a fixed menu of suggestions, where my selection from that list is used to update the "status" value in the underlying file. 

Any ideas on how I can modify this segment of the code to achieve the desired result? I tried asking around with folks who are more familiar with Dataview, but no one was able to help. I thought maybe someone more familiar with Templater would have ideas. 

Thank you!

**Comment by Zachatoo** [comment]
Have you tried using `autoprop` from the [MetaEdit API](https://github.com/chhoumann/MetaEdit#api) instead of a Templater prompt/suggester? MetaEdit has a built in suggester, through autoprop, should be much easier than doing the suggester part through Templater.
